src/Stratis.Bitcoin.Features.Wallet.Tests/WalletSyncManagerTest.cs

```diff
@@ -357,5 +363,35 @@ public void SetWalletTip(ChainedHeader tip)
                 this.walletTip = tip;
             }
         }
+
+        private static void WaitLoop(Func<bool> act, string failureReason, int millisecondsTimeout = 50)
+        {
+            if (failureReason == null)
+                throw new ArgumentNullException(nameof(failureReason));
+
+            var cancel = new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 30 * 1000);
+            while (!act())
+            {
+                try
+                {
+                    cancel.Token.ThrowIfCancellationRequested();
+                    Thread.Sleep(millisecondsTimeout);
+                }
+                catch (OperationCanceledException e)
+                {
+                    Assert.False(true, $"{failureReason}{Environment.NewLine}{e.Message}");
+                }
+            }
+        }
+
+        private uint256 ExpectedBlockHash(IWalletSyncManager walletSyncManager, int blockHeight)
```

Comment:
Maybe change the name to `AssertTipBlockHash` or something.

Reply:
ok